packages/tools/src/tools/displayTools/Surface/addOrUpdateSurfaceToElement.ts

```diff
@@ -25,6 +25,16 @@ function addOrUpdateSurfaceToElement(
 
   const surfaceActor = surfaceActorEntry?.actor as Types.Actor;
 
+  const isVisible = surface.visible;
+
+  if (!isVisible) {
+    if (surfaceActor) {
+      viewport.removeActors([surfaceActorEntry.uid]);
```

Comment:
Try using `actor.setVisibility(true)` at the actor level instead of `remove`; there should be a `setVisibility` API there.